pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -578,20 +579,28 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         """
         if self.dirpath is not None:
             return  # short circuit
-
-        if trainer.logger is not None:
+        if trainer.loggers:
             if trainer.weights_save_path != trainer.default_root_dir:
                 # the user has changed weights_save_path, it overrides anything
                 save_dir = trainer.weights_save_path
-            else:
+            elif len(trainer.loggers) == 1:
                 save_dir = trainer.logger.save_dir or trainer.default_root_dir
+            else:
+                save_dir = trainer.default_root_dir
 
+            name = (
+                trainer.loggers[0].name
+                if len(trainer.loggers) == 1
+                else "_".join(dict.fromkeys(str(logger.name) for logger in trainer.loggers))
```

Comment:
```suggestion
                "_".join(str(logger.name) for logger in trainer.loggers)
```
can you add a test for both?